`jdatetime.datetime.isoformat` fails on python 2.7 when called after `from __future__ import unicode_literals`

```python
$ python2
Python 2.7.13 (default, Apr 16 2021, 14:02:03)
[GCC 6.3.0 20170516] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import jdatetime
>>> jdatetime.datetime(1398, 4, 11, 11, 6, 5, 123456).isoformat('T')
u'1398-04-11T11:06:05.123456'
>>> from __future__ import unicode_literals
>>> jdatetime.datetime(1398, 4, 11, 11, 6, 5, 123456).isoformat('T')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/mnt/nfs/labstore-secondary-tools-home/5j9/venv2/local/lib/python2.7/site-packages/jdatetime/__init__.py", line 1269, in isoformat
    'argument 1 must be a single character: {}'.format(sep)
AssertionError: argument 1 must be a single character: T
>>>

```

I believe the code above should work, even after the `__future__` import; similar to the way, for example, `jdatetime.datetime(1398, 4, 11, 11, 6, 5, 123456).strftime('%Y')` works.

# Please include your runtime information

* jdatetime version: 3.7.0
* Python version: 2.7
* OS: Linux

--- a/jdatetime/__init__.py
+++ b/jdatetime/__init__.py
@@ -39,8 +39,10 @@ timestamp_is_supported = (
 
 if sys.version_info[0] >= 3:  # py3
     _int_types = (int,)
+    _basestring = str
 else:
     _int_types = (int, long)  # noqa
+    _basestring = basestring  # noqa: F821
 
 if platform.system() == 'Windows':
     FA_LOCALE = 'Persian_Iran'
@@ -1235,11 +1237,11 @@ class datetime(date):
             return self.tzinfo.dst(self)
         return None
 
-    def isoformat(self, sep=str('T'), timespec='auto'):
+    def isoformat(self, sep='T', timespec='auto'):
         """[sep] -> string in ISO 8601 format,
         YYYY-MM-DDTHH:MM:SS[.mmmmmm][+HH:MM]."""
 
-        assert isinstance(sep, str) and len(sep) == 1, \
+        assert isinstance(sep, _basestring) and len(sep) == 1, \
             'argument 1 must be a single character: {}'.format(sep)
 
         tz = self.strftime("%z")

--- a/t/test.py
+++ b/t/test.py
@@ -686,6 +686,14 @@ class TestJDateTime(unittest.TestCase):
 
         self.assertAlmostEqual(jiso, '1398-04-11M00:00:00')
 
+    def test_isoformat_unicode_arg_python2(self):
+        jdt = jdatetime.datetime(1398, 4, 11)
+        jiso = jdt.isoformat('M')
+        # Used to raise:
+        # AssertionError: argument 1 must be a single character: M
+        ujiso = jdt.isoformat(u'M')
+        self.assertEqual(jiso, ujiso)
+
     def test_isoformat_bad_sep(self):
         jdt = jdatetime.datetime(1398, 4, 11)
 

